pkg/controllers/cache/cache.go

```diff
@@ -248,7 +248,7 @@ func (jc *jobCache) DeletePod(pod *v1.Pod) error {
 		return err
 	}
 
-	if jc.jobs[key].Job == nil {
+	if jobTerminated(job) {
```

Comment:
What is the purpose of modifying this?

Reply:
A job is associated with multiple pods, we only clean up the job after all pods are deleted. Judging the `jobTerminated ` method in advance can reduce the number of retries